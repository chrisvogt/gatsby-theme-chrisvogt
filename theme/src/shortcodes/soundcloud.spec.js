import React from 'react'
import renderer from 'react-test-renderer'
import SoundCloud from './soundcloud'

describe('SoundCloud Shortcode', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<SoundCloud soundcloudId='880888540' />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a default title if one is not provided', () => {
    const testRenderer = renderer.create(<SoundCloud soundcloudId='880888540' />)
    const testInstance = testRenderer.root
    expect(testInstance.findByType('iframe').props.title).toEqual('Song on SoundCloud')
  })
})
