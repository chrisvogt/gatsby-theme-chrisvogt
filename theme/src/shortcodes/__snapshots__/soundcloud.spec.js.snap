// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoundCloud Shortcode matches the snapshot in dark mode 1`] = `
<iframe
  allow="autoplay"
  frameborder="no"
  height="166"
  scrolling="no"
  src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/880888540&color=%23800080&auto_play=false&hide_related=false&show_comments=true&show_user=true&show_reposts=false&show_teaser=true"
  title="Song on SoundCloud"
  width="100%"
/>
`;

exports[`SoundCloud Shortcode matches the snapshot in light mode 1`] = `
<iframe
  allow="autoplay"
  frameborder="no"
  height="166"
  scrolling="no"
  src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/880888540&color=%23ff5500&auto_play=false&hide_related=false&show_comments=true&show_user=true&show_reposts=false&show_teaser=true"
  title="Song on SoundCloud"
  width="100%"
/>
`;
